src/server/frontend_wayland/wayland_default_configuration.cpp

```diff
@@ -29,32 +29,60 @@
 namespace mf = mir::frontend;
 namespace mo = mir::options;
 
-std::shared_ptr<mf::Connector>
-    mir::DefaultServerConfiguration::the_wayland_connector()
+namespace
+{
+auto const wl_shell     = "wl_shell";
+auto const xdg_shell    = "xdg_wm_base";
+auto const xdg_shell_v6 = "zxdg_shell_v6";
+
+auto configure_wayland_extensions(std::string extensions, bool x11_enabled) -> std::unique_ptr<mf::WaylandExtensions>
 {
     struct WaylandExtensions : mf::WaylandExtensions
     {
-        WaylandExtensions(bool x11_enabled) : x11_enabled{x11_enabled} {}
+        WaylandExtensions(std::set<std::string> const& extension, bool x11_enabled) :
+            extension{extension}, x11_enabled{x11_enabled} {}
+
     protected:
         virtual void custom_extensions(
             wl_display* display,
             std::shared_ptr<mf::Shell> const& shell,
             mf::WlSeat* seat,
             mf::OutputManager* const output_manager)
         {
-            if (!getenv("MIR_DISABLE_XDG_SHELL_V6_UNSTABLE"))
-                add_extension("zxdg_shell_v6", std::make_shared<mf::XdgShellV6>(display, shell, *seat, output_manager));
+            if (extension.find(wl_shell) != extension.end())
```

Comment:
Another extension could, theoretically, have a name that contains `"wl_shell"`.
```
if ((":" + extension + ":").find(":" + wl_shel +":") != extension.end())
```
Would be uglier, but more technically correct IMO

Reply:
There's probably also a leaner way to do this.

Reply:
@wmww That code is looking for a value "wl_shell" in a set. It doesn't match a value containing "wl_shell".

Reply:
Oh, oops. I thought I read that carefully.